epoxy-viewbinder/src/main/java/com/airbnb/epoxy/EpoxyViewBinder.kt

```diff
@@ -0,0 +1,549 @@
+ *
+ * @param initializer a lambda that is run directly after instantiation of the
+ * [LifecycleAwareEpoxyViewBinder].
+ *
+ * @return a view binder or null if a view with the [viewId] could not be found.
+ */
+fun ViewGroup.optionalEpoxyView(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    fallbackToNameLookup: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit = { },
+    modelProvider: ModelCollector.(Context) -> Unit
+) = lazy {
+    val view = this
+    // View id is not present, we just return null in that case.
+    if (view.maybeFindViewByIdName<View>(viewId, fallbackToNameLookup) == null) return@lazy null
+
+    return@lazy epoxyViewInternal(
+        viewId = viewId,
+        useVisibilityTracking = useVisibilityTracking,
+        fallbackToNameLookup = fallbackToNameLookup,
+        initializer = initializer,
+        modelProvider = modelProvider
+    )
+}
+
+private fun ComponentActivity.epoxyViewInternal(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    fallbackToNameLookup: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit,
+    modelProvider: ModelCollector.(Context) -> Unit
+) = LifecycleAwareEpoxyViewBinder(
+    this,
+    { findViewById(android.R.id.content) },
+    viewId,
+    useVisibilityTracking = useVisibilityTracking,
+    fallbackToNameLookup = fallbackToNameLookup,
+    modelProvider = modelProvider
+).apply(initializer)
+
+private fun Fragment.epoxyViewInternal(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    fallbackToNameLookup: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit,
+    modelProvider: ModelCollector.(Context) -> Unit
+) = LifecycleAwareEpoxyViewBinder(
+    viewLifecycleOwner,
+    { view },
+    viewId,
+    useVisibilityTracking = useVisibilityTracking,
+    fallbackToNameLookup = fallbackToNameLookup,
+    modelProvider = modelProvider
+).apply(initializer)
+
+private fun ViewGroup.epoxyViewInternal(
+    @IdRes viewId: Int,
+    useVisibilityTracking: Boolean = false,
+    fallbackToNameLookup: Boolean = false,
+    initializer: LifecycleAwareEpoxyViewBinder.() -> Unit,
+    modelProvider: ModelCollector.(Context) -> Unit
+) = LifecycleAwareEpoxyViewBinder(
+    (this.context.getContextForSharedViewPool() as? LifecycleOwner)
+        ?: error("LifecycleOwner required as view's context "),
+    { this },
+    viewId,
+    useVisibilityTracking = useVisibilityTracking,
+    fallbackToNameLookup = fallbackToNameLookup,
+    modelProvider = modelProvider
+).apply(initializer)
+
+/**
+ * Attempts to find this view's parent Activity in order to share the view pool. If this view's
+ * `context` is a ContextWrapper it will continually unwrap it until it finds the Activity. If
+ * no Activity is found it will return the the view's context.
+ */
+private fun Context.getContextForSharedViewPool(): Context {
```

Comment:
could rename this to `unwrapContextForLifecycle`.

Should the comment be updated since this no longer is used for view pools?